packages/react-components/src/shared/src/omitProps.ts

```diff
@@ -0,0 +1,10 @@
+export function omitProps<TObject extends Record<string, any>, TProps extends keyof TObject>(obj: TObject, props: TProps[]): Omit<TObject, TProps> {
+    return (Object.keys(obj) as TProps[]).
+        reduce((acc, x) => {
+            if (!props.includes(x)) {
+                acc[x] = obj[x];
+            }
+
+            return acc;
+        }, {} as TObject);
+}
```

Comment:
The complexity of this thing explode in TS.

Does "props" could be a Record<string, any> ? Would make this simpler.

Something like:

```
export function omitProps(obj: Record<string, any>, props: Record<string, any>) => Record<string, any> {
   ....
}
```

Reply:
while simpler to read, this does not leverage typescript.

const testObject = { 
variable1: true,
variable2: false,
}



using my method : 
```
const result = omit(testObject, ["variable1"])
result.variable2 // typescript does not allow this
result.variable1 // this is a bool
```

using a generic Record<string, any> : 
```
const result = omit(testObject, ["variable1"])
result.variable2 // no warning here
result.variable1 // this is type any
```

Reply:
I've created a sandbox for this case

https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCm0DyAjAVvYsC8MDeMAUAG4CGATgJakoA28AjAFxzkCu8ANCRdXfACYWAM1K0IXQgF9CheAA8ADiHKxhbMFkrgYIALaUoABVJQAFCHQsASphUATADzQqYAOacYpMAE8AfJ6K5CCKECwulO4A2gC6AJQ2duROEe6e3v74hDA5uTDk8FBs5GAwZqgYWAB0ANbwPhAW6HFxVdl5uQX2bMDwZmakwMCe8nEw2H5ZHdMwlMJlAIRBIRBVkcC0bPaIZqNjeO0zM4PAUfIx47roZzEA3IdHMDIPRwVFJV5D90dSnnhSXggMFsoGSzigrg8Xl8kzi92eCmUqhg6k0UG0pX0hgAgvR5I4ACoVTCwBQIMD2IEghzgyHpGGeAlGYKhGBk+AUoF1HwgeZE9AkvxNNAsfmVKCBFlhGBMqWxBIwJAGKCE4lYRnMlaTA55N7FUrlAXVbmNSxoMakIGylbytqveDdXr9E4jMYTKaPHJzRbLUJrTSbbaNPYez15E43S5mm7fMPPMP5Qr6z7AWPTX74AGWmVqqBw6SyUCQWAFCBsWjGUxR5UmcwIZBGiVRABEZCoNHoDGb8XupfLlagVTbfE7txgAHpx9CfJ4wCA2eRguRCH2K7Wh7wO4Ix5Pp7P5-BFypC+BoImyxXcQpqzi8WZ61Bc54W8Ot12eyvEP2r-IN+3+AwE5ToeS6fheUA-n+I6CEBcA+Io8AwCgIAgPQ3hAA